### ISSUE TITLE
Add parser support for SQLite's STRICT tables

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14631]
SQLite 3.37.0 finally introduced `STRICT` tables, which allow for ignoring SQLite's historic type affinity JavaScript-ed-ness, which jOOQ has always ignored:
https://www.sqlite.org/stricttables.html

It is reasonable to expect users to make heavy use of this modifier in the future. Given that jOOQ never really cared about the type affinity feature and always pretended SQLite was strict like all the others, we can simply parse and ignore this flag for now.

This will be useful:

- In the `DDLDatabase`
- Whenever the `SQLiteDatabase`, including to read check constraints or type information on views or virtual tables (see #8278, #11172)

Failure example here:
https://github.com/jOOQ/jOOQ/issues/9711#issuecomment-1426200463

**Comment by lukaseder** [comment]
The error is:

```
[INFO] Cannot parse SQL: CREATE TABLE t_14631 (
  i INT PRIMARY KEY,
  j INT,

  CONSTRAINT t_14631_chk_j CHECK (j > 0)
) STRICT
org.jooq.impl.ParserException: Unexpected clause: [6:3] ...NT,

  CONSTRAINT t_14631_chk_j CHECK (j > 0)
) [*]STRICT
    at org.jooq.impl.DefaultParseContext.exception (ParserImpl.java:14615)
    at org.jooq.impl.DefaultParseContext.done (ParserImpl.java:14801)
    at org.jooq.impl.DefaultParseContext.lambda$parseQuery0$2 (ParserImpl.java:961)
    at org.jooq.impl.DefaultParseContext.wrap (ParserImpl.java:14808)
    at org.jooq.impl.DefaultParseContext.parseQuery0 (ParserImpl.java:961)
    at org.jooq.impl.ParserImpl.parseQuery (ParserImpl.java:778)
    at org.jooq.impl.ParserImpl.parseQuery (ParserImpl.java:773)
    at org.jooq.meta.sqlite.SQLiteDatabase.loadCheckConstraints (SQLiteDatabase.java:333)
    at org.jooq.meta.AbstractDatabase.lambda$getRelations0$27 (AbstractDatabase.java:3124)
    at org.jooq.meta.AbstractDatabase.onError (AbstractDatabase.java:3758)
    at org.jooq.meta.AbstractDatabase.getRelations0 (AbstractDatabase.java:3124)
    at org.jooq.meta.AbstractDatabase.lambda$getRelations$18 (AbstractDatabase.java:2796)
    at org.jooq.meta.AbstractDatabase.onError (AbstractDatabase.java:3758)
    at org.jooq.meta.AbstractDatabase.getRelations (AbstractDatabase.java:2796)
    at org.jooq.meta.DefaultColumnDefinition.getPrimaryKey (DefaultColumnDefinition.java:154)
    at org.jooq.meta.AbstractTableDefinition.getPrimaryKey (AbstractTableDefinition.java:108)
    at org.jooq.codegen.JavaGenerator.generateTable (JavaGenerator.java:5969)
    at org.jooq.codegen.JavaGenerator.generateTables (JavaGenerator.java:5948)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:645)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:600)
    at org.jooq.codegen.JavaGenerator.generate0 (JavaGenerator.java:512)
    at org.jooq.codegen.AbstractGenerator.generate (AbstractGenerator.java:191)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:220)
    at org.jooq.codegen.DefaultGenerator.generate (DefaultGenerator.java:1)
    at org.jooq.codegen.GenerationTool.run0 (GenerationTool.java:957)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:240)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:235)
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:216)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:578)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

**Comment by lukaseder** [comment]
The `STRICT` keyword can't appear in `CREATE TABLE AS SELECT`:

<img width="585" alt="image" src="https://user-images.githubusercontent.com/734593/218672565-b6233269-94ca-4ee9-96c7-4efb47944821.png">

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.9 (https://github.com/jOOQ/jOOQ/issues/14632), and 3.16.15 (https://github.com/jOOQ/jOOQ/issues/14633)

**Comment by lukaseder** [comment]
Will also add support for parsing and ignoring `WITHOUT ROWID`: https://github.com/jOOQ/jOOQ/issues/14690
